### PR TITLE
refactor(oomstore/push-processor): refactor processor with group snapshot interval

### PR DESCRIPTION
**Pull request by lianxmfor**

### CODE REVIEW

**Review by wfxr — APPROVED** [review]
LGTM. (update the related tests as well?)

File: pkg/oomstore/push_processor.go [hunk 1]
```diff
@@ -209,6 +210,6 @@ func (p *PushProcessor) newRevision(ctx context.Context, s *OomStore, groupID in
 	return nil
 }
 
-func (p *PushProcessor) lastRevision(unixMill int64) int64 {
-	return (unixMill / p.revisionInteval.Milliseconds()) * p.revisionInteval.Milliseconds()
+func (p *PushProcessor) lastRevision(revisionInterval int64, unixMill int64) int64 {
```
Comment (jinghancc) [inline pkg/oomstore/push_processor.go:213]: snapshotInterval or revisionInterval

Comment (lianxmfor) [reply pkg/oomstore/push_processor.go:213]: Currently, I have written `revisionInterval`. Do you think `snapshotInterval` is better?
Mostly I'm not quite sure which one to use here, both make sense.

**Comment by lianxmfor** [comment]
> 

Yeah, I'm going to update the snapshot + cdc integration-test after this PR.

**Comment by lianxmfor** [comment]
> LGTM. (update the related tests as well?)

Yeah, I'm going to update the snapshot + cdc integration-test after this PR.

File: pkg/oomstore/push_processor.go (hunk 1 above)
Comment (lianxmfor) [reply pkg/oomstore/push_processor.go:213]: Do you mean to modify `lastRevision` to `snapshotInterval` or `revisionInterval` ?

Comment (jinghancc) [reply pkg/oomstore/push_processor.go]: I notice that line 146 uses `SnapshotInterval`, but here uses `revisionInterval`, would be better to consolidate

Comment (lianxmfor) [reply pkg/oomstore/push_processor.go]: both changed to `SnapshotInterval `

File: pkg/oomstore/push_processor.go [hunk 2]
```diff
@@ -209,6 +210,6 @@ func (p *PushProcessor) newRevision(ctx context.Context, s *OomStore, groupID in
 	return nil
 }
 
-func (p *PushProcessor) lastRevision(unixMill int64) int64 {
-	return (unixMill / p.revisionInteval.Milliseconds()) * p.revisionInteval.Milliseconds()
+func (p *PushProcessor) lastRevision(snapshotRevision int64, unixMill int64) int64 {
```
Comment (jinghancc) [inline pkg/oomstore/push_processor.go:213]: hmm typo?

**Review by jinghancc — APPROVED** [review]
LGTM

File: pkg/oomstore/push_processor.go (hunk 2 above)
Comment (lianxmfor) [reply pkg/oomstore/push_processor.go]: 🥶 fixed

**Pull request merged**